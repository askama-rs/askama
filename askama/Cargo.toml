[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
