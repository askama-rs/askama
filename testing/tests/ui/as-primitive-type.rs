use askama::Template;

#[derive(Template)]
#[template(source = r#"{{ 1234 as 4567 }}"#, ext = "html")]
struct A;

#[derive(Template)]
#[template(source = r#"{{ 1234 as ? }}"#, ext = "html")]
struct B;

#[derive(Template)]
#[template(source = r#"{{ 1234 as u1234 }}"#, ext = "html")]
struct C;

#[derive(Template)]
#[template(source = r#"{{ 1234 as core::primitive::u32 }}"#, ext = "html")]
struct D;

#[derive(Template)]
#[template(source = r#"{{ 1234 as int32_t }}"#, ext = "html")]
struct E;

#[derive(Template)]
#[template(source = r#"{{ (1234 + 4 * 12 / 45675445 - 13) as int32_t }}"#, ext = "html")]
struct F;

fn main() {
}
