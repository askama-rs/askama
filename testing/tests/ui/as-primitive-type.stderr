error: `as` operator expects the name of a primitive type on its right-hand side, not a path or alias
 --> <source attribute>:1:8
       "as 4567 }}"
 --> tests/ui/as-primitive-type.rs:4:21
  |
4 | #[template(source = r#"{{ 1234 as 4567 }}"#, ext = "html")]
  |                     ^^^^^^^^^^^^^^^^^^^^^^^

error: `as` operator expects the name of a primitive type on its right-hand side, not a path or alias
 --> <source attribute>:1:8
       "as ? }}"
 --> tests/ui/as-primitive-type.rs:8:21
  |
8 | #[template(source = r#"{{ 1234 as ? }}"#, ext = "html")]
  |                     ^^^^^^^^^^^^^^^^^^^^

error: `as` operator expects the name of a primitive type on its right-hand side, found `u1234`
 --> <source attribute>:1:8
       "as u1234 }}"
  --> tests/ui/as-primitive-type.rs:12:21
   |
12 | #[template(source = r#"{{ 1234 as u1234 }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^

error: `as` operator expects the name of a primitive type on its right-hand side, not a path or alias
 --> <source attribute>:1:8
       "as core::primitive::u32 }}"
  --> tests/ui/as-primitive-type.rs:16:21
   |
16 | #[template(source = r#"{{ 1234 as core::primitive::u32 }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `as` operator expects the name of a primitive type on its right-hand side, found `int32_t`
 --> <source attribute>:1:8
       "as int32_t }}"
  --> tests/ui/as-primitive-type.rs:20:21
   |
20 | #[template(source = r#"{{ 1234 as int32_t }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `as` operator expects the name of a primitive type on its right-hand side, found `int32_t`
 --> <source attribute>:1:35
       "as int32_t }}"
  --> tests/ui/as-primitive-type.rs:24:21
   |
24 | #[template(source = r#"{{ (1234 + 4 * 12 / 45675445 - 13) as int32_t }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
